### PR TITLE
Configuration for DUNE-FD-HD full 10kt with hydra option

### PR DESCRIPTION
**Pull request by HaiwangYu**
ref: https://indico.fnal.gov/event/65318/

### CODE REVIEW

**Review by absolution1 — COMMENTED** [review]
Hi @HaiwangYu 
Thanks for sending this over.  I've attached comments.

Some other items:
 - a CMakeLists.txt file missing in dune10kt-hd
 - a line needs adding to the CMakeLists.txt in DUNEWirecell to add the dune10kt-hd folder to the compilation
 
 Finally, I've run a quick test with a simple particle gun muon in the 10kt geom.  The muon passes through 5 of the TPC volumes.  I am seeing peak memory usage of 6.4GB.  Is that an expected amount?

**Comment by tomjunk** [comment]
still under review?

**Comment by HaiwangYu** [comment]
@tomjunk  I will make some changes to address @absolution1's comments this weekend.

**Comment by HaiwangYu** [comment]
@absolution1 , 

> Finally, I've run a quick test with a simple particle gun muon in the 10kt geom. The muon passes through 5 of the TPC volumes. I am seeing peak memory usage of 6.4GB. Is that an expected amount?

I saw similar number. See page 8 and 9 of this talk.
[2024-05-13_hydra3-full-hd.pdf](https://github.com/user-attachments/files/16745208/2024-05-13_hydra3-full-hd.pdf)

You can see from page 8 that 6APA of HD also uses 6GB resident mem.

Do you think more optimizations are needed?

**Comment by absolution1** [comment]
Hi @HaiwangYu and @tomjunk (apologies for the delay, I was on holiday)

This is OK to merge.

I do think we need to explore further memory optimisations, but that shouldn't stop this PR going in.

**Comment by aolivier23** [comment]
Looks like we have @absolution1 's approval but we need to hit the "review" button.  I'm going to review this and try to merge it Friday.  Please leave a message here if you want me to wait.

File: dunereco/DUNEWireCell/dune10kt-hd/params.jsonnet [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+                        response: centerline + res_plane,
+                        cathode: centerline + cpa_plane, 
+                    },
+                    {
+                        anode: centerline - apa_plane,
+                        response: centerline - res_plane,
+                        cathode: centerline - cpa_plane, 
+                    }
+                ],
+            } for n in std.range(0,11)], // twelve anodes
```
Comment (absolution1) [inline dunereco/DUNEWireCell/dune10kt-hd/params.jsonnet:91]: This sounds like a setup for the 1x2x6.  Is this line important?

File: dunereco/DUNEWireCell/dune10kt-hd/params.jsonnet [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        // rectangular solid.  Again "wirecell-util wires-info" helps
+        // to choose something.
+        bounds : {
+          tail: wc.point(-363.376, -600.019, -0.87625, wc.cm),
+          head: wc.point( 363.376,  600.019, 1393.46, wc.cm),
+        }
+    },
+
+    daq: super.daq {
+        nticks: 6000,
```
Comment (absolution1) [inline dunereco/DUNEWireCell/dune10kt-hd/params.jsonnet:106]: Does this get overwritten later on?  If we change the number of ticks in LArSoft will thie line supersede the larsoft value?

**Review by absolution1 — COMMENTED** [review]
Thank for checking in on this @aolivier23 
I never actually did take a look.   I have now.  I did have a couple of small questions.  Once addressed, this can be merged.
@HaiwangYu could you take a quick look?

File: dunereco/DUNEWireCell/dune10kt-hd/params.jsonnet (hunk 2 above)
Comment (tomjunk) [reply dunereco/DUNEWireCell/dune10kt-hd/params.jsonnet:106]: Is this still under review?  I think we have a general issue with LArSoft plugins that do not share common configurations.  There is a manual step needed to align the configurations across domains owned by different software components.  We had a discussion about this in the future frameworks requirements workshop, but I don't think we had a good solution.

Comment (absolution1) [reply dunereco/DUNEWireCell/dune10kt-hd/params.jsonnet:106]: I've pinged Haiwang to take a look.
We've had some success in combatting this in FD, by linking the params between config with `@local`.  That obviously only works when both configs are fcl-based.

**Comment by HaiwangYu** [comment]
Hi @absolution1 , I updated my branch for this PR. Could you take a look? Thanks!

**Review by absolution1 — COMMENTED** [review]
Thanks @HaiwangYu 
The changes look good.  Before approving, can I just confirm where far detector dimensions were taken from?  The GDML?

**Comment by HaiwangYu** [comment]
Hi @absolution1 , very good question.

First, the x (drift) positions are configured in the jsonnet.
The yz positions are configured by the wire boundaries from the gdml.

So, for the jsonnet configurations, we only consider the x-axis dimensions.

For that, I think current dimensions are derived from hd1x2x6 and adjusted by the gdml.
Comparing these two files, I think the only parameter adjusted is the `apa_cpa`
```
1x2x6: 3.63075*wc.m
fullhd: 3.635265*wc.m
```
https://github.com/DUNE/dunereco/blob/cf74225fa944ac60a56e41679ded241a0c636daf/dunereco/DUNEWireCell/dune10kt-hd/simparams.jsonnet
https://github.com/DUNE/dunereco/blob/develop/dunereco/DUNEWireCell/dune10kt-1x2x6/params.jsonnet
[2024-05-13_hydra3-full-hd.pdf](https://github.com/user-attachments/files/17437668/2024-05-13_hydra3-full-hd.pdf)


After this adjustment, the wire positions are now synchronized to the gdml:
page 3-6 of this talk:


[2024-05-13_hydra3-full-hd.pdf](https://github.com/user-attachments/files/17437687/2024-05-13_hydra3-full-hd.pdf)

**Comment by absolution1** [comment]
Hi @HaiwangYu 
Thank you for the answer!

I'm happy to approve this.

My only concern is that there's a small amount of manual adjustment needed if we change the FD geometry (which we almost certainly will do).
Is there an option where the x-axis is also calculated from the GDML?

**Comment by aolivier23** [comment]
@HaiwangYu we're still missing the CMakeLists.txt in dune10kt-hd.  Do you happen to have it locally?  I'll try to write one that just installs all of those files after lunch.

**Pull request merged**